applications/nrf_desktop/src/modules/dfu.c

```diff
@@ -171,6 +186,16 @@ static void terminate_dfu(void)
 	if (IS_ENABLED(CONFIG_CAF_POWER_MANAGER_EVENTS)) {
 		power_manager_restrict(MODULE_IDX(MODULE), POWER_MANAGER_LEVEL_MAX);
 	}
+
+	if (IS_ENABLED(CONFIG_DESKTOP_DFU_LOCK) &&
+	    !k_work_delayable_is_pending(&background_erase) &&
+	    !k_work_delayable_is_pending(&reboot_request)) {
+		int err;
+
+		err = dfu_lock_release(&config_channel_owner);
+		__ASSERT_NO_MSG(!err);
+		ARG_UNUSED(err);
```

Comment:
Then we would be sure that lock would not be released prematurely

Reply:
> checks for all of the pending works and assertion on the returned value).

It is not possible to do that because you need to have different pending work checks based on the current execution context (in the background erase you cannot check if background erase is pending)